src/compiler/utils.jl

```diff
@@ -360,6 +360,46 @@ function reinsert_gcmarker!(func::LLVM.Function, @nospecialize(PB::Union{Nothing
     end
 end
 
+@inline enum_attr_kind(kind::String) = LLVM.API.LLVMGetEnumAttributeKindForName(kind, Csize_t(length(kind)))
+
+const swiftself_kind = enum_attr_kind("swiftself")
+
+Base.@assume_effects :removable :foldable :nothrow function has_swiftself(fn::LLVM.Function)::Bool
```

Comment:
Why those effects? We shouldn't add them to every function, and I kinda doubt they apply here or are useful?